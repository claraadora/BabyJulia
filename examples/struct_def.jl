abstract type Mammal end 

struct Cat <: Mammal 
    name::String
    age::Int64
    isFemale::Bool
end 

struct Woman <: Mammal
    name::String
    height::Int64
    isAlive::Bool	
end

kitty = Cat("Kitty", 1, true)
obama = Woman("Michelle Obama", 180, true)

println(kitty.name)
println(kitty.age)
println(kitty.isFemale)
obama
