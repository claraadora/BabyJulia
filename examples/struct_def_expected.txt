{"name":"Michelle Obama","height":180,"isAlive":true}